src/stashcache.py

```diff
@@ -178,19 +179,24 @@ def _cache_is_allowed(resource, vo_name, stashcache_data, public, suppress_error
     return ret
 
 
-def generate_cache_authfile(vo_data: VOsData, resource_groups: List[ResourceGroup], fqdn=None, legacy=True, suppress_errors=True) -> str:
+def generate_cache_authfile(global_data: GlobalData,
```

Comment:
Do you need to give similar treatment to `generate_public_cache_authfile` also, to handle the `if public_only` block here?

https://github.com/opensciencegrid/topology/blob/e2d34ad34579f974c5467dc93617d8bd299fe30a/src/app.py#L253-L256

Reply:
Yup! I'll get right on it